src/main/java/org/pawpal/controller/PetController.java

```diff
@@ -0,0 +1,42 @@
+package org.pawpal.controller;
+
+import org.pawpal.dto.PetDTO;
+import org.pawpal.model.Pet;
+import org.pawpal.service.PetService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/pets")
+public class PetController {
```

Comment:
pentru toate api urile adauga niste comenturi care sa explice ce asteapta sa primeasca si ce returneaza:
/**
 * Retrieves a pet's details based on the given ID.
 *
 * @param id the unique identifier of the pet, provided in the URL path.
 * @return PetDTO containing the details of the specified pet.
 * @throws PetNotFoundException if no pet is found with the given ID.
 *
 * Example:
 * <pre>
 *     GET /pets/1
 *     Response: {
 *       "id": 1,
 *       "name": "Buddy",
 *       "type": "Dog",
 *       "age": 3
 *     }
 * </pre>
 */